create-litr.Rmd

```diff
@@ -1228,12 +1230,53 @@ add_chunk_label_hyperlinks <- function(html_files,
           )
       }
     )
-
-    writeLines(txt, con = html_files[i])
+    
+    parsed_html <- xml2::read_html(paste(txt,collapse="\n"))
+    span_nodes <- xml2::xml_find_all(parsed_html, ".//span[contains(., '###')]")
+    for(j in seq_along(span_nodes)){
+      span_node <- span_nodes[[j]]
+      span_id <- xml2::xml_attr(span_node, "id")
+      span_node_path <- stringr::str_split(xml2::xml_path(span_node),"/")
+      pre_path <- paste(span_node_path[[1]][1:(length(span_node_path[[1]])-2)],collapse="/")
+      pre_parent <- xml2::xml_find_first(parsed_html, pre_path)
+      # add parent nodes to pre_parent
+      xml2::xml_add_parent(pre_parent,xml2::read_xml('<div class="custom-field"> </div>'))
+      outer_div <- xml2::xml_parent(pre_parent)
+      xml2::xml_add_child(outer_div, xml2::read_xml(stringr::str_glue('<h1><span id="{span_id}">{span_id}</span></h1>')),.where = 0)
+      xml2::xml_add_parent(pre_parent,xml2::read_xml('<div> </div>'))
+      # remove span with id in comment form last
+      xml2::xml_remove(span_node)
+      # remove the extra line break that is left over from removing the span
+      code_node <- xml2::xml_child(pre_parent)
+      # remove the first line break
+      if(length(xml2::xml_contents(code_node)) > 1){
+        xml2::xml_remove(xml2::xml_contents(code_node)[1])
+      } else {
+        changed_txt <- stringr::str_remove(paste(as.character(xml2::xml_contents(code_node)),collapse=""), "\n")
+        xml2::xml_replace(code_node, xml2::read_xml(stringr::str_glue('<code>{changed_txt}</code>')))  
+      }
+    }
+    
+    # last thing is to insert an additional style node in the head with our CSS
+    css_string <- ".custom-field {border: 4px solid;border-top: none;padding: 8px; padding-bottom: 0px; margin-bottom: 10px; margin-top: 10px}
+.custom-field h1 {font: 16px normal;margin: -16px -8px 0;font-family:inherit;font-weight:500;line-height:1.1;color:inherit}
+.custom-field h1 span {float: left;}
+.custom-field h1:before {border-top: 4px solid;content: ' ';float: left;margin: 8px 2px 0 -1px;width: 12px;}
+.custom-field h1:after {border-top: 4px solid;content: ' ';display: block;height: 24px;left: 2px;margin: 0 1px 0 0;overflow: hidden;position: relative;top: 8px;}"
+    head_node <- xml2::xml_find_first(parsed_html, ".//head")
+    xml2::xml_add_child(head_node, xml2::read_xml(stringr::str_glue("<style type='text/css'>{css_string}</style>")))
+    txt <- xml2::write_html(parsed_html, html_files[i])
```

Comment:
Perhaps this could be wrapped in a function (perhaps a function internally defined within the function, like `find_chunk_defs()` was earlier in this same function)?  I feel like this is a quite large chunk of code all with the goal of doing one thing (adding that formatting).

Reply:
Also, I would add to the roxygen description of `add_chunk_label_hyperlinks()` the fact that this formatting occurs.